### ISSUE TITLE
link-grammar is has new dependency on zlib-dev

### ISSUE THREAD

**Issue by linas** [opencog/link-grammar#457]
Old scripts are failing, because the new link-grammar configure script fails because zlib-dev is missing. But LG does not need zlib-dev for anything...!?

**Comment by ampli** [comment]
The minisat headers include it. However, the LG library don't use anything that needs zlib.
I will check how this dependency can be removed, at least with the bundled library.
Maybe zlib.h can just be included in the distribution? (Its license is permissive.)

**Comment by linas** [comment]
I created a quick hacky emergency fix for version 5.3.13

I would rather not start including zlib-dev as part of the distribution. This another step down the path of DLL-hell.

In fact, I'm tempted to move in the opposite direction: remove our bundled minisat. It got bundled only because 5-6 years ago, minisat was not available in any distro. Now that it is available in pretty much all of them, I think, , we should stop bundling it.

**Comment by ampli** [comment]
For Mac I could still not locate a minisat2 library package. There is a minisat2 package for it (under the "homebrew" package manager), but it installs only a statically linked minisat application.

For Windows I also could not locate a minisat binary package. BTW, I didn't try to compile LG+sat-solver on Windows using the current bundled library (with the previous one there were compilation problems that I had never cared to investigate.)

So it seems to me the solution is just to ifdef-out zlib dependency in the bundled minisat.

**Comment by linas** [comment]
yeah, I can't imagine why zlib is even needed. What does compression have to do with any of this?

**Comment by ampli** [comment]
It seems it is for reading a compressed equation file (for the minisat provided application).
There are only a few lines of zlib dependency and I can just remove/modify them.
I can send a PR.
